config/routes.rb

```diff
@@ -1,22 +1,24 @@
 require "sidekiq/web"
 
 Rails.application.routes.draw do
-  devise_for :users
-
-  resource :home, only: :show
-  resource :about, only: :show, controller: :about
-  resource :conduct, only: :show
-  resource :pricing, only: :show, controller: :pricing
-  resource :role, only: :new
-  resources :businesses, except: :destroy
-  resources :read_notifications, only: :index, path: "/notifications/read"
-  resources :notifications, only: %i[index show]
-  resources :conversations, only: %i[index show] do
-    resources :messages, only: :create
-    resource :block, only: %i[new create]
-  end
-  resources :developers, except: :destroy do
-    resources :messages, only: %i[new create], controller: :cold_messages
+  scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
```

Comment:
```suggestion
  scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/, defaults: {locale: nil} do
```

This change should fix the weirdness with the missing optional parameters when using route helpers. I think we can also get rid of some of that `default_url` configuration, too, but I haven't tested that.

Reply:
This is pretty neat! 👍
I have redo related changes and force pushed.

Reply:
I have removed this setting because all url helpers in View will get `locale nil`.
Instead, the `resolve_locale` in `ApplicationController` will help us to set the locale to nil when it's the default locale.